lib/lan_connection.js

```diff
@@ -38,80 +40,71 @@ LANConnection.prototype.end = function() {
-
+function findConnections() {
+  var lanConfig, envConfig;
   // Grab all mDNS connected Tessels
-  findmDNSConnections(function(err, connections) {
-    // Riase any errors
-    if (err) {
-      return callback && callback(err);
-    }
-
-    // Check if a config file exists
-    fs.exists(__dirname + '/../config.env', function (exists) {
-      // If it does
-      if (exists) {
-        // parse it
-        config = envfile.parseFileSync(__dirname + '/../config.env')
-        // Create a LAN Connection with the provided auth credentials
-        var conn = new LANConnection({
-          host: config.host,
-          username: 'root',
-          port: 22,
-          privateKey: fs.readFileSync(config.keyPath),
-          passphrase: config.keyPassphrase
+  return findmDNSConnections()
+    // TODO: Remove this step once Tessel Setup is built
+    .then(function(tessels){
+      return new Promise(function(resolve, reject) {
+        // Check if a config file exists
+        fs.exists(__dirname + '/../config.env', function (exists) {
+          // If it does
+          if (exists) {
+            // parse it
+            envConfig = envfile.parseFileSync(__dirname + '/../config.env')
+            // Create a LAN Config with the provided auth credentials
+            lanConfig = {
+              host: envConfig.host,
+              username: 'root',
+              port: 22,
+              privateKey: fs.readFileSync(envConfig.keyPath),
+              passphrase: envConfig.keyPassphrase
+            }
+          } else {
+            return reject('Set up your config.env with keyPath and keyPassphrase');
+          }
+          // Pass the tessel's through or else you have no tessels
+          return resolve(tessels);
         });
-
-        // Insert the specified connection as the first item
-        // so that it will be preferred
-        connections.unshift(conn);
-      }
-
-      // Open the connections and return them
-      return openConnections(connections, callback);
-    });
-  });
-
-  // For each LAN Conenction, try to open up an SSH session
-  // Should we really be doing this here?
-  function openConnections(connections, callback) {
-    async.each(connections, function(conn, cb) {
-      conn.open(function(err) {
-        conn.initError = err;
-        cb(err);
       });
-    }, function(err) {
-      callback(err, connections)
-    });
-  }
+    })
+    // Create new LAN connections for each tessel.
+    .map(function(tessel){
+      if (tessel.host === lanConfig.host){
+        return new LANConnection(lanConfig);
+      }
+      // Add credentials before testing?
+      return new LANConnection(tessel);
+    }, {concurrency: 6})
+    // Test the connections.
+    // Not sure if we want to throw errors here or somewhere else.
+    .map(function(connection){
+      return connection.open();
+    }, {concurrency: 6});
 }
 
-// TODO: Actually find Tessels on the network using mDNS
 function findmDNSConnections(callback) {
```

Comment:
Can we remove (refactor) this function? It's only got one line and it just returns another function.